core/src/com/unciv/models/translations/Translations.kt

```diff
@@ -31,16 +31,32 @@ class Translations : LinkedHashMap<String, TranslationEntry>(){
 
     private var modsWithTranslations: HashMap<String, Translations> = hashMapOf() // key == mod name
 
-
-    fun get(text: String, language: String, activeMods: HashSet<String>? = null): String {
+    /**
+     * Searches for the translation entry of a given [text] for a given [language].
+     * This includes translations provided by mods from [activeMods]
+     *
+     * @param text the input text for the translate entry
+     * @param language the inquired language
+     * @param activeMods set of the active mods that should include in the search
+     *
+     * @return the translation entry or null when not available
+     */
+    fun get(text: String, language: String, activeMods: HashSet<String>? = null): TranslationEntry? {
         activeMods?.forEach {
             modsWithTranslations[it]?.let { modTranslations ->
-                val translation = modTranslations[text]?.get(language)
-                if (translation != null) return translation
+                val translationEntry = modTranslations[text]?.get(language)
+                if (translationEntry != null) return modTranslations[text]
             }
         }
 
-        return this[text]?.get(language) ?: text
+        return this[text]
+    }
+
+    /**
+     * @see get
+     */
+    fun getText(text: String, language: String, activeMods: HashSet<String>? = null): String {
+        return get(text, language, activeMods)?.get(language) ?: text
```

Comment:
Why are you sending the language into the get() function, checking if the language exists inside the get() function, but returning only the entry and not the language?